### PR TITLE
[tests-only] wait for services in drone acceptance tests

### LINKED ISSUE #39561: [tests-only] wait for mailhog to start before running acceptance tests
https://drone.owncloud.com/owncloud/core/33924/165/17
```
cURL error 6: Could not resolve host: email (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://email:8025/api/v2/messages (GuzzleHttp\Exception\ConnectException)
```

https://drone.owncloud.com/owncloud/core/33924/165/4

The pull of mailhog was very slow, and the tests started before it came up. (docker `mailhog/mailhog` runs with host name `email` in CI)

Add a wait-for email in drone.

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
- wait for the mailhog email service to be available before starting acceptance tests (this was the reason for starting this PR - we had some pipeline fails because of not waiting)
- refactor other existing wait-for steps to use `owncloudci/wait-for:latest` (like done in oCIS, web etc)
- refactor `.drone.star` to put the docker image names in constants at the top:
```
OC_CI_ALPINE = "owncloudci/alpine:latest"
OC_CI_CORE_NODEJS = "owncloudci/core:nodejs14"
OC_CI_WAIT_FOR = "owncloudci/wait-for:latest"
OC_CI_NODEJS = "owncloudci/nodejs:14"
OC_UBUNTU = "owncloud/ubuntu:20.04"
```
like done in oCIS, web etc.

## Related Issue
- Fixes #39561 

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
example of wait-for email:
https://drone.owncloud.com/owncloud/core/34049/118/12

example of wait-for selenium (browser):
https://drone.owncloud.com/owncloud/core/34049/126/14

example of wait-for server:
https://drone.owncloud.com/owncloud/core/34049/126/12

File: .drone.star [hunk 1]
```diff
@@ -2055,16 +2059,16 @@ def browserService(browser):
 
     return []
 
-def waitForBrowserService(phpVersion, isWebUi):
-    if isWebUi:
+def waitForBrowserService(browser):
+    if (browser == "chrome") or (browser == "firefox"):
```
Comment (kiranparajuli589) [inline .drone.star:2063]:
```suggestion
    if browser in ["chrome", "firefox"]:
```

**Comment by phil-davis** [comment]
I converted to draft because this change to `.drone.star` is not urgent. So I will wait until next week to merge it. I don't want to risk complicating the `release-10.9.0` branch CI by dropping a drone CI change into master just now,

But feel free to review my code.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm

**Comment by phil-davis** [comment]
I rebased to latest master just now to get an up-to-date CI result.
This is ready for review and merge.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by kiranparajuli589 — APPROVED** [review]

**Pull request merged**